### ISSUE TITLE
Namespace issues

### ISSUE THREAD

**Issue by joeytakeda** [projectEndings/diagnostics#7]
How do we deal with different namespace in documents? Currently, we use the indiscriminate element selector * with no namespace prefix. There were some erroneous statistics and links being found in a project that had <egXML>s taken from the TEI guidelines in their documentation. In our templates, do we care about the parent element's namespace when we index @xml:ids or when we iterate through the attributes?

For example: 

1. Is it necessary for references made in the egXML namespace to be locatable within the rest of the project?  Example:

`<egXML xmlns="http://www.tei-c.org/ns/Examples" xml:id="example1"><p>This is valid text that references the Julian date <date calendar="#Julian">1600</date>.</p><egXML>`

If nothing is declared with an @xml:id='Julian' in the TEI namespace, should this be an error? 

2. Can you link across namespaces? Example:
`<egXML xmlns="http://www.tei-c.org/ns/Examples" xml:id="example1"><p>This is valid text.</p><egXML>`
[...]
`<p>As we can see from the <ref target="#example1">above example</ref></p>`

Is this an error?

**Comment by martindholmes** [comment]
Good point. We have:

`xpath-default-namespace="http://www.tei-c.org/ns/1.0"`

but then we go and undermine it by doing this:

~~~~~
<xsl:for-each select="//@*[not(local-name(.) = $excludedAtts)]">
~~~~~

You're right that we shouldn't check the egXML namespace; that makes no sense. But we presumably want to allow pointing between elements in different namespaces, so perhaps we should have an excluded namespace list, and put egXML in it.

**Comment by martindholmes** [comment]
Fix committed in commit e1a0b38.
